src/useValidation/useValidation.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { mount } from '@vue/test-utils'
+import { describe, expect, it, vi } from 'vitest'
+import { nextTick, ref } from 'vue'
+import { ValidationRule, useValidation } from '@/useValidation/useValidation'
+
+function flushPromises(): Promise<void> {
+  return new Promise(setImmediate)
+}
+
+const isGreaterThanZero: ValidationRule<number> = (value, name) => {
+  if (value > 0) {
+    return true
+  }
+
+  return `${name} must be greater than 0`
+}
+
+describe('useValidation', () => {
+  describe('validate', () => {
+    it('sets valid to true when the rules pass', async () => {
+      const { valid, error, validate } = useValidation(
+        ref(1),
+        'Number',
+        isGreaterThanZero,
+      )
+
+      await validate()
+
+      expect(valid.value).toBe(true)
+      expect(error.value).toBe('')
+    })
+
+    it('sets valid to false with an error message when the rules do not pass', async () => {
+      const { valid, error, validate } = useValidation(
+        ref(0),
+        'Number',
+        isGreaterThanZero,
+      )
+
+      await validate()
+
+      expect(valid.value).toBe(false)
+      expect(error.value).toBe('Number must be greater than 0')
+    })
+
+    it('triggers automatically when the value changes', async () => {
+      const theValue = ref(0)
+      const validationRule = vi.fn().mockResolvedValue('Validation did not pass')
+      const wrapper = mount({
+        setup() {
+          const { valid, error } = useValidation(
+            theValue,
+            validationRule,
+          )
+
+          return { valid, error }
+        },
+      })
+
+      expect(wrapper.vm.valid).toBe(true)
+      expect(wrapper.vm.error).toBe('')
+      expect(validationRule).not.toHaveBeenCalled()
+
+      theValue.value = 1
+      await nextTick()
+      await flushPromises()
```

Comment:
ah yeah I noticed this after.. a bit. I suppose i should remove all of the nextTick calls since to your point it's not quite right. when it does work, it's probably happen-stance but originally I didn't realize that. In the past i've seen/used [flush-promises](https://www.npmjs.com/package/flush-promises) but rather than add it, I stumbled onto this rather simple implementation (from [here](https://rickschubert.net/blog/posts/flushing-promises/)). 

i didn't see the other timeout utility. Hmmm tbh idk off-hand the difference but I can look into it a bit more and/or remove in favor of the existing timeout utility.

Reply:
I was just curious if you knew something about nextTick that I didn't. Cause I can never get that to work like it seems it should based on the description in the vue docs.

Reply:
i really thought nextTick was working for... longer than I expected 😅 (well the tests passed!), but I guess it makes sense that effects run off a separate schedule entirely from the batched DOM updates that nextTick is i guess more for...

Reply:
Even when I'm working with the dom nextTick never works lol.